AzureDevOpsPipelines/deployment/bot-publish/source/core.js

```diff
@@ -0,0 +1,96 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const tl = require("azure-pipelines-task-lib");
+const child_process_1 = require("child_process");
+const path = require("path");
+const fs = require("fs");
+class core {
+    constructor() {
+        this.isLoggedIn = false;
+        this.cliPasswordPath = "";
+        this.servicePrincipalId = "";
+        this.servicePrincipalKey = "";
+        this.prefix = "";
+        this.subscriptionID = "";
+        this.getPrefix();
+        this.subscriptionID = tl.getVariable('subscriptionID');
+        if (this.subscriptionID == null) {
+            this.loginAzureRM(tl.getInput("connectedServiceNameARM", true));
+            tl.setVariable('subscriptionID', this.subscriptionID);
+        }
+    }
+    loginAzureRM(connectedService) {
+        var authScheme = tl.getEndpointAuthorizationScheme(connectedService, true);
+        this.subscriptionID = tl.getEndpointDataParameter(connectedService, "SubscriptionID", true);
+        if (authScheme.toLowerCase() == "serviceprincipal") {
+            let authType = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
+            let cliPassword;
+            var servicePrincipalId = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
+            if (authType == "spnCertificate") {
+                tl.debug('certificate based endpoint');
+                let certificateContent = tl.getEndpointAuthorizationParameter(connectedService, "servicePrincipalCertificate", false);
+                cliPassword = path.join(tl.getVariable('Agent.TempDirectory') || tl.getVariable('system.DefaultWorkingDirectory'), 'spnCert.pem');
+                fs.writeFileSync(cliPassword, certificateContent);
+                this.cliPasswordPath = cliPassword;
+            }
+            else {
+                tl.debug('key based endpoint');
+                cliPassword = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+                this.servicePrincipalId = servicePrincipalId;
+                this.servicePrincipalKey = cliPassword;
+            }
+            var tenantId = tl.getEndpointAuthorizationParameter(connectedService, "tenantid", false);
```

Comment:
If it's value is only set once here, it'd be a const instead of var. Also, can we put it on top with all the variables declarations? If not, add a space between lines